src/GenAPI.Desktop/Microsoft.DotNet.BuildTools.GenAPI.targets

```diff
@@ -0,0 +1,33 @@
+﻿<?xml version="1.0" encoding="utf-8" ?>
+<Project ToolsVersion="12.0" xmlns="http://schemas.microsoft.com/developer/msbuild/2003">
+  <PropertyGroup>
+    <GenAPITargetPath Condition="'$(GenAPITargetPath)' == ''">$(TargetDir)$(TargetName).cs</GenAPITargetPath>
```

Comment:
I didn't see a need for an extra intermediate file in this case.  Indeed, the goal was for folks to commit the file.  If someone wants to build a "direct to dll" package that imports this they can, overriding the path to the file and calling CSC in that target.